tests/unit/test_jenkins.py

```diff
@@ -250,6 +251,87 @@ def test__setup_user_token_raises_exception(mock_env: jenkins.Environment):
             jenkins.Jenkins(mock_env)._setup_user_token(mock_container)
 
 
+@pytest.mark.parametrize(
+    "exception",
+    [
+        pytest.param(requests.exceptions.JSONDecodeError, id="decode_error"),
+        pytest.param(KeyError, id="keyerror"),
+    ],
+)
+def test__setup_user_token_security_disabled_response_parse_error(
```

Comment:
Note that ideally you should only test public APIs

Reply:
This test is only for the workaround that I mentioned above and should be removed with the refactor follow-up PR.